returnn/dataset.py

```diff
@@ -90,3 +93,56 @@ def run(self):
 
             self.out_mean.set(total_mean)
             self.out_std_dev.set(numpy.sqrt(total_var))
+
+
+class SpeakerLabelHDFFromBlissJob(Job):
+    """
+    Extract speakers from a bliss corpus and create an HDF file with a speaker index
+    matching the speaker entry in the corpus speakers for each segment
+    """
+
+    def __init__(self, bliss_corpus, returnn_root=None):
+        """
+        :param bliss_corpus: bliss XML corpus where the speakers and segments are taken from
+        :param returnn_root: used to import SimpleHDFWriter from a specific path if not installed in the worker env
+        """
+        self.bliss_corpus = bliss_corpus
+        self.returnn_root = returnn_root
+        self.out_speaker_hdf = self.output_path("speaker_labels.hdf")
+        self.out_num_speakers = self.output_var("num_speakers")
+        self.out_speaker_dict = self.output_path("speaker_dict.pkl")
+
+    def tasks(self):
+        yield Task("run", mini_task=True)
+
+    def run(self):
+        bliss = corpus.Corpus()
+        bliss.load(self.bliss_corpus.get_path())
+        speaker_by_index = {}
+        index_by_speaker = {}
+        num_speakers = len(bliss.speakers)
+        self.out_num_speakers.set(num_speakers)
+        # speakers are stored as OrderedDict, so this is a safe operation
+        for i, speaker in enumerate(bliss.all_speakers()):
+            speaker_by_index[i] = speaker.name
+            index_by_speaker[speaker.name] = i
+
+        pickle.dump(speaker_by_index, uopen(self.out_speaker_dict, "wb"))
+
+        SimpleHDFWriter = get_returnn_simple_hdf_writer(
+            returnn_root=self.returnn_root.get_path() if self.returnn_root else None
+        )
+        hdf_writer = SimpleHDFWriter(
+            self.out_speaker_hdf.get_path(), dim=num_speakers, ndim=1
+        )
+
+        for recording in bliss.all_recordings():
+            for segment in recording.segments:
+                speaker_name = segment.speaker_name or recording.speaker_name
+                speaker_index = index_by_speaker[speaker_name]
+                segment_name = "/".join([bliss.name, recording.name, segment.name])
```

Comment:
what if there are sub-corpora? Better use the existing fullname function

```suggestion
                segment_name = segment.fullname()
```

Reply:
good catch